pkg/kubelet/pod/pod_manager.go

```diff
@@ -153,6 +158,9 @@ func (pm *basicManager) UpdatePod(pod *v1.Pod) {
 // lock.
 func (pm *basicManager) updatePodsInternal(pods ...*v1.Pod) {
 	for _, pod := range pods {
+		if pm.secretManager != nil {
```

Comment:
@timstclair - IIUC, you are asking for the update case, right? For this case we are storing current version of pod in the secretStore itself. If this is not nil (which is the case in case of creation), we are removing old secrets based on this pod. Please see secretStore for it.